packages/nimble-components/src/table/models/keyboard-navigation-manager.ts

```diff
@@ -0,0 +1,1243 @@
+import { Notifier, Subscriber, Observable } from '@microsoft/fast-element';
```

Comment:
One thing I noticed (and this may not be new), is that if you have a cell focused, particularly one that has something like an action menu in it and press `Enter`, there is no visible state change, but you have have exited navigation mode, and arrow keys no longer do anything. Additionally, a successive press of `Enter` doesn't active the action menu.

The latter part may be intentional, but it seems odd that after you press `Enter`, you a) can't get to the interactive elements via the arrow keys, and b) there's no visual indication that you are in a new mode.

To me, the proper behavior is what is described [here in the ARIA guidance](https://www.w3.org/WAI/ARIA/apg/patterns/grid/#keyboardinteraction-settingfocusandnavigatinginsidecells), the important part of which is:

- `Enter`: Disables grid navigation and:
    -    If the cell contains editable content, places focus in an input field, such as a [textbox](https://w3c.github.io/aria/#textbox). If the input is a single-line text field, a subsequent press of Enter may either restore grid navigation functions or move focus to an input field in a neighboring cell.
    -    If the cell contains one or more widgets, places focus on the first widget.
    
So, in essence, if a cell is focused, and the only interactive element is an action menu, and a user presses `Enter` I would think the action menu should be focused. If a cell has no interactive elements at all, I would propose that a keypress of `Enter` be considered a no-op (though there is no direct ARIA guidance for this afaik).

Reply:
Good catch. I'd definitely say the current behavior is incorrect - I don't think there should be any case where there's no visual focus change, but that you're no longer in navigation mode.

I think it's debatable whether Enter should focus the action menu button or not. It's a bit of a special case (and not really editable content as referenced in that ARIA guidance).
However, if it doesn't focus the menu button, I think it should be a no-op (i.e. you should still be in cell navigation mode).

Reply:
Actually once I looked at the code, what you described (action menu button gets focused) was the intent, I just had a typo (`<=` instead of `<` in a bounds check).

Pushed an update to fix that, and added a test.

Reply:
Sweet! Also, just to clarify, the ARIA guidance doesn't demand that the content be editable. The second bullet simply states that if the cell has one or more widget, that it places focus on the first widget. This is what I think describes the action menu use-case.